### DISCUSSION TITLE
`-extend` may make the transparent background turn to white

### DISCUSSION THREAD

**Discussion by RFYoung** [ImageMagick/ImageMagick#4926] [Development]
Hello, I was using [wand](https://github.com/emcconville/wand), a binding of ImageMagick on Python. I encountered a [problem](https://github.com/emcconville/wand/issues/568), which may be seen as an upstream issue from ImageMagick.

Here is a small example to show the issue:
``` shell
magick -size 100x100 xc:transparent -font 'Roboto-Regular.ttf' -pointsize 56 -fill blue -draw 'text 0 60 G'  -extent 150x90 demo_bug.png 
```
Here, the background would turn to white. A workaround is to add `-background transparent` before `-extent`.
So is this actually designed to be like this, or is this an issue?
Thanks !

PS: I am sorry that I am very new to ImageMagick and may ask stupid problem. :joy:

**Comment by GeeMack** [comment]
Generally when using `-extent` to increase the size of the canvas the entire added background will be filled with the `-background X` color **and** that will affect the areas that are already transparent, like this...

    magick -size 480x160 -background none xc:blue xc:none xc:red -append -background green -extent 110% resultA.png

![resultA](https://user-images.githubusercontent.com/65475279/157473694-0bfa3897-96bf-43bf-8e9d-098bfa09c031.png)

 You can change that behavior by setting the compose method to "copy" before the `-extent` operation like this...

    magick -size 480x160 -background none xc:blue xc:none xc:red -append -background green -compose copy -extent 110% resultB.png

![resultB](https://user-images.githubusercontent.com/65475279/157473741-a331ca64-1eda-4282-bb23-60405377040b.png)

If you have any following `-extent`, `-append`, or `-composite` operations you should change that compose method back to "over" or whatever you need before the next operation.

**Reply by RFYoung** [reply]
Thank you for your reply.
I am sorry that I think I need some time to learn all the relevant  knowledge about the reply above.
Anyway, thanks again!

**Comment by snibgo** [comment]
As GeeMack says.

See also the "-extent" documentation at https://imagemagick.org/script/command-line-options.php#extent which starts with:
```
If the image is enlarged, unfilled areas are set to the background color.
```

**Reply by RFYoung** [reply]
Thanks for you reply !
I am sorry but I still a bit confused. So the 'transparent' background is actually something unfilled? Or is there some action in my command cause the background unset?

**Comment by fmw42** [comment]
xc:"transparent" is different from -background transparent.  xc: defines the color for a new canvas or image.  -background is a setting for a number of commands such as -extent.  Setting xc: to transparent does not set the background color to transparent.  The default background color is white.
